### DISCUSSION TITLE
Security, how to rate limit and limit capabilities for clients?

### DISCUSSION THREAD

**Discussion by reprojectacc** [soketi/pws#386] [General]
Hello,

is there a way to rate limit clients f.e. based on their IP, or other metrics?
Is there a way to force clients to authenticate?

Also how do I limit the capabilities for clients
- to subscribe to public channels
- to send client events

Thank you.

**Comment by rennokki** [comment]
The app is not meant to rate limit by IP or other metrics, but you can easily proxy the request through NGINX (for example) and use the built-in rate limiting: https://www.nginx.com/blog/rate-limiting-nginx/

The only limits are based on the maximum size limits and throughput of backend/client events and are imposed at the app-level. Take a look at the default driver for app definitions: https://docs.soketi.app/app-management/array-driver
